cnxpublishing/tests/views/test_admin.py

```diff
@@ -334,6 +339,53 @@ def test_admin_content_status_bad_page_number(self):
             admin_content_status(request)
         self.assertIn('invalid page', caught_exc.exception.message)
 
+    @mock.patch('celery.result.AsyncResult')
+    @mock.patch('psycopg2.connect')
+    def test_admin_content_status_state_icons(self, mock_psycopg2_connect,
+                                             mock_async_result):
+        states = ['PENDING', 'QUEUED', 'STARTED', 'RETRY', 'SUCCESS',
+                  'FAILURE', 'REVOKED', 'UNKNOWN']
+        def round_robin_states(*args, **kwargs):
+            result = mock.Mock(
+                state=states[mock_async_result.call_count - 1 % len(states)])
+            result.failed.return_value = False
+            return result
+
+        mock_async_result.side_effect = round_robin_states
+
+        cursor = mock.MagicMock()
+        cursor.fetchall.return_value = [
+            ('Just some random module {} for testing'.format(i),
+             ['authors'],
+             uuid.uuid4(),
+             'print-style',
+             'latest-recipe',
+             'latest-recipe',
+             '1.1',
+             '1.1',
+             datetime.now().isoformat(),
+             'result-{}'.format(i))
+            for i in range(len(states))]
+
+        mock_psycopg2_connect.return_value.__enter__().cursor().__enter__.return_value = cursor
```

Comment:
😆  That's a hell of a line. Maybe need to add a `# noqa ...` to it? http://flake8.pycqa.org/en/latest/user/violations.html#in-line-ignoring-errors

Reply:
Oh oops... forgot to fix it... :rofl: